signer/src/storage/postgres.rs

```diff
@@ -247,6 +247,66 @@ impl super::DbRead for PgStore {
         .map_err(Error::SqlxQuery)
     }
 
+    async fn get_pending_accepted_deposit_requests(
+        &self,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: i32,
+        threshold: i64,
+    ) -> Result<Vec<model::DepositRequest>, Self::Error> {
+        sqlx::query_as!(
+            model::DepositRequest,
+            r#"
+            WITH RECURSIVE context_window AS (
+                -- Anchor member: Initialize the recursion with the chain tip
+                SELECT block_hash, block_height, parent_hash, created_at, 1 AS depth
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+                
+                UNION ALL
+                
+                -- Recursive member: Fetch the parent block using the last block's parent_hash
+                SELECT parent.block_hash, parent.block_height, parent.parent_hash,
+                       parent.created_at, last.depth + 1
+                FROM sbtc_signer.bitcoin_blocks parent
+                JOIN context_window last ON parent.block_hash = last.parent_hash
+                WHERE last.depth < $2
```

Comment:
Is this supposed to be `<=`? We might need to change this and thw query in `get_pending_deposit_requests` or the query in `get_pending_withdraw_requests`.

Reply:
I do think this is correct, since we're joining the parent which has `last.depth + 1`. We also have tests ensuring this is consistent with the in memory implementation, which iterates over `(0..context_window)`. Let me know if you still believe this to be incorrect, and we can investigate this further.

Reply:
Yeah, you're right, this is fine. It gets `context_window` blocks. And the query in `get_pending_withdraw_requests` is fine as well since we explicitly want to go past the context window. Never mind 😅!